ios-app/YouPlay/Services/PlaylistService/PlaylistServiceImpl/PlaylistServiceImpl.swift

```diff
@@ -148,6 +148,40 @@ class PlaylistServiceImpl: PlaylistService {
         
         return playlistIdToSongs
     }
+
+    func addSongToLikedSongs(uid: String, song: Song) async {
+        let playlistRef = FirestoreConstants.PlaylistsCollection(uid: uid)
+        
+        do {
+            let querySnapshot = try await playlistRef.whereField("title", isEqualTo: "Liked Songs").getDocuments()
```

Comment:
Since your looking for a playlist named "Liked songs" make sure that users are unable to create playlists that already exists OR at least they just can't make a playlist named "Liked songs" that way we can avoid duplicates.